nucypher/characters/lawful.py

```diff
@@ -141,6 +141,23 @@ def grant(self, bob, uri, m=None, n=None, expiration=None, deposit=None, handpic
         policy.enact(network_middleware=self.network_middleware)
         return policy  # Now with TreasureMap affixed!
 
+    def revoke(self, policy):
+        """
+        Goes through the RevocationKit in Policy and revokes re-encryption
+        access to Bob from each arrangement.
+        """
+        # Sign the RevocationNotices in the RevocationKit in preparation for
+        # sending to Ursula.
+        policy.revocation_kit.sign(self.stamp)
+
+        failed_revocations = list()
+        for node_id, notice in policy.revocation_kit:
+            ursula = self.known_nodes[node_id]
+            response = self.network_middleware.revoke_arrangement(ursula, notice)
+            if response.status_code != 200:
+                failed_revocations.append(notice)
+
+        return failed_revocations
```

Comment:
Still an open question... What does Alice do with failed revocations (if there are some)?
Of course, an immediate option is to do nothing. Another option is to use them later. Or, she can hand them to someone to submit on her behalf later...

Anyway, that's not an immediate thing to solve in this PR

Reply:
The idea is that you serialize failed revocations and read them at startup to attempt revocation the next time you run the CLI.